### ISSUE TITLE
How to load GLTF files with texture?

### ISSUE THREAD

**Issue by chloesun** [bimspot/xeokit-react-demo#1]
Thanks for creating the react component and sharing with the community! 
From the example, this is how you reference the src
```
export const sampleModel1 = {
  id: 'duplex',
  src: './models/duplex/scene.gltf',
  metaModelSrc: './metaModels/duplex/metaModel.json',
  edges: true
};
```
For the GLTF files with texture, (gltf, bin, png), how to load them?
![spec](https://user-images.githubusercontent.com/13730281/59463045-06132c80-8df3-11e9-846b-36bdf4e98875.PNG)

**Comment by xeolabs** [comment]
@chloesun the [GLTFLoaderPlugin](https://xeokit.github.io/xeokit-sdk/docs/class/src/plugins/GLTFLoaderPlugin/GLTFLoaderPlugin.js~GLTFLoaderPlugin.html) has a ````performance```` configuration - this is normally set ````true```` to load large models into the viewer for high-performance non-realistic engineering-style rendering without textures.

Set that ````false```` to load models into xeokit's  lower-performance flavour of scene representation, that supports glTF 2 materials with physically-based rendering, textures etc.

Note that's going to freak your browser out if you try to load a hugely detailed model with that configuration. Best for a few thousand objects, not good for hundreds of thousands.

````javascript
const gltfLoader = new GLTFLoaderPlugin(viewer);

const model = gltfLoader.load({
        id: "myModel",
        src: "./models/gltf/schependomlaan/scene.gltf",
        metaModelSrc: "./metaModels/schependomlaan/metaModel.json",
        
        performance: false // <<--------- Set false to enable full materials
    });
````

**Comment by chloesun** [comment]
Here is where my GLTF stored, right under `public` directory:
![1](https://user-images.githubusercontent.com/13730281/59530711-4e455400-8eb2-11e9-99cf-72fa657a1830.PNG)

I added this in `model.js`
```
export const sampleModel3 = {
  id: 'cbr_room',
  src: './659s_gltf/web_001.gltf',
  performance: false
};
```

I got a blank canvas with the error messages like this, note I have no problem rendering models with gltf and json 
![3](https://user-images.githubusercontent.com/13730281/59530915-be53da00-8eb2-11e9-8924-b37a404d1d35.PNG)

What could be wrong?

**Comment by eriadam** [comment]
Usually "unexpected token < in JSON at position 0" means that the server gives you back a 404 error and an html response — instead of JSON.  Maybe the path to the gltf file is not correct or you don't have read permissions?

**Comment by chloesun** [comment]
@eriadam I think the path is correct, and I can render this model at https://gltf-viewer.donmccurdy.com/, so I think there is no permission issue either...

**Comment by eriadam** [comment]
@chloesun In your web inspector look into the resources being loaded and the network logs, you should see the request for the gltf file and the response as well. I am pretty sure the response is not a proper gltf.
Note that the JSON loader throws the error, not the renderer.

**Comment by chloesun** [comment]
@eriadam thanks for the speedy response, the status code is 304
![network](https://user-images.githubusercontent.com/13730281/59535055-eac12380-8ebd-11e9-993e-1ff006bc0ba1.PNG)

**Comment by eriadam** [comment]
@chloesun Cool, what is in the Preview/Response tabs?

**Comment by chloesun** [comment]
@eriadam 
response:
![response](https://user-images.githubusercontent.com/13730281/59535766-c0706580-8ebf-11e9-989a-22c11c7e6122.PNG)
preview
![preview](https://user-images.githubusercontent.com/13730281/59535774-cbc39100-8ebf-11e9-8660-c708ffc3c3cf.PNG)

**Comment by eriadam** [comment]
Yap, it's html alright. When you request `http://localhost:1234/cbr_room/cbr.gltf` even simply from the browser, your local server needs to return the gltf file. I am not sure about your local setup, maybe some reverse proxy or web server config issue.

**Comment by barnabasmolnar** [comment]
@chloesun 

Are you by any chance using Parcel to serve your app? I see your css and js files have these hashes that I know Parcel generates when bundling your app, although other bundlers may do that too, of course. In any case, @eriadam is almost certainly right in that the gltf file is not being served properly. If you're using a bundler like Parcel that serves static files with these hashes, then your gltf file should have one too.

I've mostly relied on create react app to serve react applications so I'm not sure I can be of much help on this one. If you are, however, willing to share a bit more on how you're bundling your app, we might at the very least be able to help point you in the right direction.

**Comment by chloesun** [comment]
I'm using Parcel to bundle the app

**Comment by barnabasmolnar** [comment]
Might be worth asking the parcel guys about this.

This is what a quick search yielded for me, not sure if this is relevant to your use-case: https://github.com/parcel-bundler/parcel/issues/1080

Maybe have a look at the last comment and the plugin mentioned there:
https://www.npmjs.com/package/parcel-plugin-static-files-copy

Honestly though, I'm only shooting in the dark here, I've never actually used Parcel to serve a React app. :blush: 
Please do let us know if you manage to figure it out.

**Comment by chloesun** [comment]
@barnabasmolnar Thanks for the guidance. I've tried this library  https://github.com/tiaanduplessis/parcel-plugin-asset-copier, to solve the servicing static files like `gltf` with Parcel, based on this discussion https://github.com/parcel-bundler/parcel/issues/1411

The JSON error is gone, and I saw this error from the console:
![index](https://user-images.githubusercontent.com/13730281/59630948-9fed1900-9114-11e9-9e75-da7ac88579d0.PNG)

Since I want to reference gltf model with `gltf` `bin` and `png`, and I think I only need to add one `gltf` src here :
```
export const sampleModel3 = {
  id: 'cbr',
  src: 'cbr_room/cbr.gltf',
  performance: false
};
```
If I remove the `performance:false` tag, there is no error, and the `gltf` and `bin` file loaded properly, but there is nothing showed in the viewer
![view](https://user-images.githubusercontent.com/13730281/59634238-4ab50580-911c-11e9-8012-389a7b72b034.PNG)






Somehow, this approach can reference the geometry, but still not the texture.
I also tested with your newly updated code, it is the same behaviour.
